src/client/request.go

```diff
@@ -98,9 +115,59 @@ func (r *Request) SetContext(ctx context.Context) *Request {
 	return r
 }
 
+// WaitFor ensures that all remaining listeners will be deferred until subRequest done
+// See TestWaitForSubRequest test
+func (r *Request) WaitFor(subRequest *Request) {
+	r.lock.Lock()
+	r.waitingFor = append(r.waitingFor, subRequest)
+	r.lock.Unlock()
+
+	subRequest.OnResponse(func(response *Response) *Response {
+		r.invokeListeners()
+		return response
+	})
+}
+
+func (r *Request) isWaiting() bool {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	for _, subRequest := range r.waitingFor {
+		if !subRequest.done {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *Request) nextListener() *ResponseListener {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	// No more listeners to invoke
+	if len(r.listeners) == 0 {
+		return nil
+	}
+
+	// Remove listener from slice
+	listener := r.listeners[0]
+	r.listeners = r.listeners[1:]
+	return listener
+}
+
 func (r *Request) invokeListeners() {
-	for _, listener := range r.listeners {
-		r.response = listener.Invoke(r.response)
+	for {
+		if r.isWaiting() {
```

Comment:
No `listeners` z main requestu sa vykonaju iba, ked su dokoncene vsetky requesty z `waitFor` ... a kedze to kontrolujeme po dokonceni kazdeho sub-requestu ... tak sa vykonaju az po poslednom.